src/com/lftechnology/java/training/niraj/ems/services/implementations/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,219 @@
+package com.lftechnology.java.training.niraj.ems.services.implementations;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.lftechnology.java.training.niraj.ems.domains.Employee;
+import com.lftechnology.java.training.niraj.ems.enums.Operators;
+import com.lftechnology.java.training.niraj.ems.enums.Roles;
+import com.lftechnology.java.training.niraj.ems.enums.Status;
+import com.lftechnology.java.training.niraj.ems.exceptions.CustomException;
+import com.lftechnology.java.training.niraj.ems.services.AbstractDao;
+import com.lftechnology.java.training.niraj.ems.utils.DbFacade;
+
+public class EmployeeDaoImpl extends AbstractDao<Employee, String> {
+
+    private static Connection conn;
+
+    public EmployeeDaoImpl() {
+        conn = DbFacade.getConnection();
+    }
+
+    @Override
+    public Employee save(Employee t) throws SQLException {
+        if (t.getId() == null) {
+            int id = insertQuery(conn, t);
+            t.setId(String.valueOf(id));
+
+        } else {
+            updateQuery(conn, t);
+        }
+        return t;
+    }
+
+    @Override
+    public Employee find(String s) throws SQLException {
+        Employee employee = new Employee();
+        Map<String, String> condition = new LinkedHashMap<String, String>();
+        condition.put(employee.getPrimaryKey(), s);
+        List<Map<String, String>> resultList = findQuery(conn, employee, condition, Operators.AND);
+        parseEmployeeObject(resultList, employee);
+        return employee;
+    }
+
+    @Override
+    public Employee findByAttributes(Map<String, String> conditions) throws SQLException {
+        Employee employee = new Employee();
+        List<Map<String, String>> resultList = findQuery(conn, employee, conditions, Operators.AND);
+        parseEmployeeObject(resultList, employee);
+        return employee;
+    }
+
+    @Override
+    public boolean update(Employee t) throws SQLException, CustomException {
+        if (t.getId() != null) {
+            return updateQuery(conn, t) > 0 ? true : false;
+        } else {
+            throw new CustomException("Invalid Employee Object");
+        }
+    }
+
+    @Override
+    public List<Employee> findAll() throws SQLException {
+        Employee employee = new Employee();
+        Employee searchResult;
+        List<Employee> employeeList = new ArrayList<Employee>();
+        List<Map<String, String>> results = findQuery(conn, employee);
+        System.out.println(results);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)